test/src/unittests/rhythm/test_loopbpmestimator.py

```diff
@@ -0,0 +1,120 @@
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from essentia.standard import MonoLoader, LoopBpmEstimator
+import numpy as np
+
+class TestLoopBpmEstimator(TestCase):
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(LoopBpmEstimator(), {'confidenceThreshold': -1})
+
+    def testRegression(self):
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))()        
+      
+        expectedEstimate= 125
+        estimate = LoopBpmEstimator(confidenceThreshold=0.95)(audio)            
+        self.assertEqual(expectedEstimate, estimate) 
+        
+        # Test for upperbound of Confidence Threshold
+        estimate = LoopBpmEstimator(confidenceThreshold=1)(audio)     
+        self.assertEqual(expectedEstimate, estimate)
+
+        # Test for lowerbound of Confidence Threshold
+        estimate = LoopBpmEstimator(confidenceThreshold=0)(audio)   
+        self.assertEqual(expectedEstimate, estimate)
+
+        # Further integrity checks at randomly chosen points in audio.
+        # If any future changes break these asserts,
+        # then this will indicates algorithm is compromised.
+
+        # Randomly chosen points, Section 1
+        # A segment of techno_loop gives a non-ideal estimation of BPM.
+        samplePoint1 = 1000
+        samplePoint2 = 6000
+        expectedEstimate = 100.0
+
+        # 0% confidence expectation triggers non-zero bpm close to estimate result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.0)(audio[samplePoint1:samplePoint2])                
+        self.assertAlmostEqual(expectedEstimate, estimate, 0.1) 
+
+        # 10% confidence expectation triggers non-zero result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.1)(audio[samplePoint1:samplePoint2])                
+        self.assertNotEqual(0, estimate) 
+
+        # Higher than 20% confidence expectation triggers non-zero bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.25)(audio[samplePoint1:samplePoint2])                
+        self.assertNotEqual(0, estimate) 
+
+        # Higher than 50% confidence expectation triggers non-zero bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.55)(audio[samplePoint1:samplePoint2])                
+        self.assertAlmostEqual(expectedEstimate, estimate, 5) 
+
+        # High confidence expectation triggers 0 bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.95)(audio[samplePoint1:samplePoint2])            
+        self.assertEqual(0, estimate) 
+
+        # Randomly chosen points, Section 2
+        # A segment of techno_loop gives a non-ideal estimation of BPM.        
+        samplePoint1 = 22000
+        samplePoint2 = 38000
+        expectedEstimate = 100.0
```

Comment:
Same here. 

We can use the same sub-segments as in the LoopBpmConfidence algorithm.